### PR TITLE
Remove microbenchmarks

### PR DESCRIPTION
**Pull request by bboreham**
These benchmarks are all testing things related to what Prometheus does, so perhaps have some historical interest, but we should not retain them in the main repo. Prometheus benchmarks should run actual Prometheus code.

(Two exceptions: `BenchmarkLabelSetAccess` is doing the same as `BenchmarkLabels_Get` in the `labels` package, and `BenchmarkLabelsClone` was moved into the `labels` package.)

As an alternative, we could retain the code but call `b.SkipNow()` at the top of each, so they don't take up any time when running all benchmarks.

### CODE REVIEW

File: tsdb/test/labels_test.go [hunk 1]
```diff
@@ -1,216 +0,0 @@
...
-// limitations under the License.
-
-package test
-
-import (
-	"bytes"
-	"crypto/rand"
-	"testing"
-
-	"github.com/prometheus/prometheus/model/labels"
```
Comment (bwplotka) [inline tsdb/test/labels_test.go:21]: Those benchmarks actually benchmark labels, so our code, no?

Comment (bboreham) [reply tsdb/test/labels_test.go:21]: OK wasn't true that Prometheus never calls `copy` on labels, but it is now I merged #11717 :-)
I guess the `labels` package could benchmark `labels.Copy()`,

Comment (bboreham) [reply tsdb/test/labels_test.go:21]: I moved the label-copying benchmark into the `labels` package.

**Review by codesome — APPROVED** [review]
I dont really have an opinion on this. So approving to unblock.

**Review by bwplotka — APPROVED** [review]
👍🏽 Thanks!

**Comment by bwplotka** [comment]
Rebase might be needed 🙈

**Pull request merged**